src/amo/api/collections.js

```diff
@@ -28,42 +32,118 @@ export const getCollectionDetail = (
 
 type GetCollectionAddonsParams = {|
   ...GetCollectionParams,
+  nextURL?: string,
   page?: number,
 |};
 
 export const getCollectionAddons = (
-  { api, page, slug, user }: GetCollectionAddonsParams
-) => {
+  { api, nextURL, page, slug, user }: GetCollectionAddonsParams
+): Promise<PaginatedApiResponse<ExternalCollectionAddon>> => {
   if (!slug) {
     throw new Error('slug is required');
   }
   if (!user) {
     throw new Error('user is required');
   }
 
-  return callApi({
+  const request = {
     auth: true,
-    endpoint: `accounts/account/${user}/collections/${slug}/addons`,
-    params: { page },
+    endpoint:
+      nextURL || `accounts/account/${user}/collections/${slug}/addons`,
+    params: undefined,
     state: api,
-  });
+  };
+  if (page) {
+    request.params = { page };
```

Comment:
Would there ever be a case where other params exist? Maybe we should be defining `params` as `{}` above and doing `request.params = { ...request.params, page };` here? It might make the above `params: undefined` code a bit easier to logically follow as well.

Reply:
> Would there ever be a case where other params exist?

No. Here's an explanation that will hopefully clarify things.

This function does not accept `request.params` as input. The line of code you're looking at is a hard coded execution of `callApi()` to handle only two cases:

1. Getting collection add-ons for a given user/collection at page 1 (the default page)
2. Getting collection add-ons at page *N* (via the `next` URL returned by the API)

Also, due to the way spreads work, `request.params` should only be defined if `page` is set (this is why it's wrapped in an if statement). Otherwise, the `page` parameter from the `next` URL will be reset. Imagine this:

```js
const query = {
  // This is how callApi() will extract ?page= from the `next` URL
  ...{ page: 2 },
  // This would be the custom `request.params` value passed to `callApi()`
  ...{ page: undefined }
};
```

In that code the value of `query.page` would be `undefined` which is not what we want. We want it to be `2`. The fix is to execute the function more like:

```js
callApi({ endpoint: nextUrlWithPageParam, params: undefined, ... })
```

Let me know if you have more questions about it.

Reply:
I'll add some comments to the code